tests/engine/reliability/test_deadlocks.py

```diff
@@ -10,7 +10,8 @@
 )
 
 
-@pytest.mark.skip(reason="Causes a deadlock.")
+@pytest.mark.service("slow")
+@pytest.mark.timeout(120)
 def test_map_wait_for_many_tasks():
```

Comment:
It looks like running these two tests is causing the only-services tests to fail. Could that be related to the limiter for SQLite?

Reply:
I think these tests are too slow on CI and are getting cancelled. The first runs in 115 seconds locally for me. I'm probably going to disable again for this pull request. They're too slow to run in CI anyway.

Reply:
I've rebased onto `main` which drops those changes as they were cherry-picked from #8903